src/plots/cartesian/axes.js

```diff
@@ -84,28 +84,72 @@ function expandRange(range) {
  *     extraOption if there is no axis)
  * extraOption: aside from existing axes with this letter, what non-axis value is allowed?
  *     Only required if it's different from `dflt`
+ * domainRef: true if ' domain' should be appended to the axis items in the list
+ *     of possible values for this axis reference.
  */
-axes.coerceRef = function(containerIn, containerOut, gd, attr, dflt, extraOption) {
+axes.coerceRef = function(containerIn, containerOut, gd, attr, dflt, extraOption, domainRef) {
```

Comment:
It appears this function is called from every place using `domainRef` true. 
@nicholas-esterer could you please confirm if that is intended?

Reply:
Yes, this is because of the previous review where it was established this would be the better way, I guess because then in the future it would be easy to make cases where `domainRef` would not be allowed.

Reply:
Not clear why we are attaching these to the `cartesian` axes objects?!
IMHO, we should drop the `domainRef` argument here and from the callers.
Then move this `coerceRef` and `addAxRefDomainCoerceRefExtra` functions to `src/components/annotations/axis_ref.js` file.

Reply:
Also this item is not resolved yet.

Reply:
> Not clear why we are attaching these to the `cartesian` axes objects?!
> IMHO, we should drop the `domainRef` argument here and from the callers.
> Then move this `coerceRef` and `addAxRefDomainCoerceRefExtra` functions to `src/components/annotations/axis_ref.js` file.

This is not correct. These functions are in the `axes` module. So let's keep where they are.

Reply:
> Yes, this is because of the previous review where it was established this would be the better way, I guess because then in the future it would be easy to make cases where `domainRef` would not be allowed.

Please only drop `domainRef` from the function argument and the callers since it has been called using `true` from everywhere. Thanks.